MozillaFoundation/foundation.mozilla.org
# Related product updates now being reflected in page previews

# Description



Link to sample test page: https://foundation-s-11799-rela-3u9dyh.herokuapp.com/cms/pages/22/edit/ 
Test page credentials: admin2/admin2
Related PRs/issues: #11799


This PR introduces a dedicated template for previewing product pages, located at `previews/buyersguide/product_page.html`. 

It overrides the `{% block related_products %}` from the standard product page template to use a new template, `related_products_section.html`, which is filled with data retrieved through a new method named `preview_related_products`.

This new method leverages the existing utility `get_related_items` to fetch related product relationships directly from the CMS's in-memory state rather than the database. 

This addresses the issue where changes to related products within the CMS would not appear in the preview pane unless the page was published.

Lastly, this PR adds some tests for the new `preview_related_products` method.



# Steps to test

1. Visit https://foundation-s-11799-rela-3u9dyh.herokuapp.com/cms/pages/22/edit/ (credentials: admin2/admin2)
2. Scroll down to the related products section
3. Make a change
4. Note that the preview panel is updating accordingly
5. If everything is working as expected, testing is complete!
